experimental/web_dashboard/lib/src/widgets/categories_dropdown.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:web_dashboard/src/api/api.dart';
+
+/// Subscribes to the latest list of categories and allows the user to select
+/// one.
+class CategoryDropdown extends StatefulWidget {
+  final CategoryApi api;
+  final ValueChanged<Category> onSelected;
+
+  CategoryDropdown({
+    @required this.api,
+    @required this.onSelected,
+  });
+
+  @override
+  _CategoryDropdownState createState() => _CategoryDropdownState();
+}
+
+class _CategoryDropdownState extends State<CategoryDropdown> {
+  Category _selected;
+  List<Category> _categories = [];
+  StreamSubscription _subscription;
+
+  void initState() {
+    super.initState();
+    _fetch();
+    _subscription = widget.api.stream().listen((categories) {
+      setState(() {
+        _categories = categories;
```

Comment:
Red screen of death... It can happen if you delete a category on another device, so I added 25041dec6bafa28746dfa4c034ded0249214dd93 to handle this case.